Python/compile.c

```diff
@@ -3641,9 +3645,15 @@ compiler_try_star_except(struct compiler *c, stmt_ty s)
         }
         if (handler->v.ExceptHandler.type) {
             VISIT(c, expr, handler->v.ExceptHandler.type);
-            ADDOP_JUMP(c, JUMP_IF_NOT_EG_MATCH, except);
+            ADDOP(c, CHECK_EG_MATCH);
+            ADDOP_I(c, COPY, 1);
```

Comment:
Since `CHECK_EG_MATCH` is always followed by this `COPY`, why not have it push a duplicate value to prepare for the jump check to avoid an extra instruction?

Reply:
This is not performance critical code, and it is quite complicated. It's not worth making it more obscure.